### PR TITLE
build rootfs with Centos7 base for arm64

### PR DESCRIPTION
**Pull request by wfurt**
This is initial effort.  In absence of debootstrap, this change builds minimal set from hand-picked RPMs. 
After that, we run yum to fix upgrade, fix dependencies and install packages we care about. 

```
sudo eng/common/cross/build-rootfs.sh  centos7 arm64
```

With this I can build corefx for arm64. (running on Ubuntu host)  I did not try to build coreclr. 

I could get llvm-5 into rootfs but it is not needed for corefx and I'm not sure if that is worth of the troubles. 

contributes to https://github.com/dotnet/corefx/issues/35838

### CODE REVIEW

**Comment by janvorli** [comment]
This looks great! It would be great to add arm32 too - I suppose it would be mostly just a matter of copying the packages file and renaming aarch64 to armhf, right?

**Comment by wfurt** [comment]
yes, this should work for arm32 as well.  I could do that but I wanted to get some feedback before final polish.

**Comment by wfurt** [comment]
arm32 added. It give me some grief until I realized I need to fake /proc/cpuinfo to select correct architecture.

**Comment by janvorli** [comment]
> until I realized I need to fake /proc/cpuinfo

I wonder if the setarch command would work to.

File: eng/common/cross/arm/centos7/epel.repo [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[epel]
```
Comment (janvorli) [inline eng/common/cross/arm/centos7/epel.repo]: Why are the repos different for arm32 and arm64?

Comment (wfurt) [reply eng/common/cross/arm/centos7/epel.repo]: technically, we do not need extra repos right now. 
I added them to make adding package easier if we need to. The SCL repro can bring clang5 and lldb but that does exist only for aarch64 and powerpc64. While looking I found EPEL for arm32 so I added as another package source. arm32 is clearly lagging in choices.

**Comment by janvorli** [comment]
@wfurt I've tried it with coreclr and it doesn't work yet. While the C compiler is detected properly, it fails detecting the C++ one. Adding the libstdc++-devel package fixed it.
The next issue is missing lttng-ust-devel package. However it seems it is not available for aarch64 from any of the repos that you have added. The build-rootfs.sh reports `No package lttng-ust-devel available`

**Comment by wfurt** [comment]
thanks @janvorli for trying. Technically, we do not need to build coreclr to solve the curl, right? 

I can c++ easily and I can probably rebuild  lttng-ust if we still care. (I think event pipe provides superior functionality for tracing)
For me the stopping point for trying to build coreclr was lldb. I can get it for arm64 but getting clang&lldb for arm32 is going to be painful.  Unless we move build of sosplugin out of coreclr very soon.

**Comment by janvorli** [comment]
It would be great to build all of our repos using the same rootfs. We don't need clang in the rootfs. But we would need the lldb library. And it is available for CentOS 7:
llvm-toolset-7-lldb-devel-5.0.1 - I've tried to add it and it got installed.

**Comment by wfurt** [comment]
I could get llvm-toolset-7-lldb-devel-5.0.1  for arm64 but not for arm32.

**Comment by janvorli** [comment]
Sigh, I was hoping that since it was there for arm64, it would be present for arm32 as well.

**Comment by wfurt** [comment]
That comes from scl not epel. It seems like arm32 is dragging behind. I do understand the motivation. 
I can check if we can steal lldb fromn some other RPM based distro - at least for arm32. Do you know what minimal version do we need?

**Comment by jonfortescue** [comment]
@janvorli @tmat has any progress been made on this so far?

**Comment by jonfortescue** [comment]
@wfurt is who I meant to tag on this, sorry. Has any progress been made?

**Comment by wfurt** [comment]
I agreed on plan with @janvorli but I did not finish that work. We no longer need lldb and that makes it much easier. The next step is to add lltng and that is hopefully last block to build coreclr as well. 
I'm on the road until 7/24 and my ability to work on this was sparse.

**Comment by jonfortescue** [comment]
@wfurt What's the status on this? Is it still needed & should we merge?

**Comment by wfurt** [comment]
As we dropped dependency on libcurl on .NET5,   https://github.com/dotnet/corefx/issues/35838 got closed.

**Pull request closed without merging**